src/unity/python/turicreate/toolkits/classifier/logistic_classifier.py

```diff
@@ -10,6 +10,7 @@
 from __future__ import division as _
 from __future__ import absolute_import as _
 import turicreate as _turicreate
+from turicreate.toolkits._main import ToolkitError
```

Comment:
You'll want to hide this import (like the other imports in this file):
`from turicreate.toolkits._main import ToolkitError as _ToolkitError`

Then in the body of your code, refer to it as `_ToolkitError` rather than `ToolkitError`.

Reply:
Because when I made the change in C++, predict_topk for multi class classification was also breaking which is not we want, right?

Reply:
> Could we add a unit test for this case?

This is already present for number of classes more than 2

```
try:
      ans =  model.predict(self.sf, output_type='probability')
except ToolkitError:
       pass
```
Should I just remove try except block because it should always in the situation raise an error

Reply:
Hmm, upon closer inspection, I believe this code is intended to handle this case:
https://github.com/apple/turicreate/blob/master/src/unity/toolkits/supervised_learning/supervised_learning.cpp#L448

We should understand why the code path doesn't go through this line, where the check should be applied and an appropriate error message already exists

Reply:
You're right that's not what we want. . . . Looking at your first change, I'm very surprised it was having that behavior. Was `log_and_throw` not actually throwing an exception?